include/rocksdb/listener.h

```diff
@@ -194,10 +194,32 @@ enum class BackgroundErrorReason {
   kManifestWriteNoWAL,
 };
 
+enum class WriteStallCause {
+  kMemtableLimit,
+  kWriteBufferManagerLimit,
+  kL0FileCountLimit,
+  kPendingCompactionBytes,
+  // Add new WriteStallCause before `kNone`
+  kNone,
+};
+
 enum class WriteStallCondition {
-  kNormal,
   kDelayed,
   kStopped,
+  // Add new WriteStallCondition before `kNormal`
+  kNormal,
+};
+
+struct CFWriteStallStatsMapKeys {
+  static const std::string& TotalStop();
+  static const std::string& TotalSlowDown();
+  static const std::string& L0FileCountLimitSlowdownsWithOngoingCompaction();
+  static const std::string& L0FileCountLimitStopsWithOngoingCompaction();
+
+  // REQUIRES: cause != `WriteStallCause::kNone`, condition !=
+  // `WriteStallCondition::kNormal`
+  static std::string CauseConditionCount(WriteStallCause cause,
+                                         WriteStallCondition condition);
 };
```

Comment:
Would it be too much if we create a .h in include just for this class and WriteStallCause, WriteStallCondition?